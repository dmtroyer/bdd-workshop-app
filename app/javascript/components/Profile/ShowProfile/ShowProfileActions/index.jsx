import React from 'react';
import { Button } from 'reactstrap';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const ShowProfileActions = ({ editOnClick }) => (
  <div data-testid="actions">
    <Button tag={Link} to="/game" color="secondary">
      Play the game
    </Button>
    <Button color="primary" onClick={editOnClick} className="float-right">
      Edit
    </Button>
  </div>
);

ShowProfileActions.propTypes = {
  editOnClick: PropTypes.func,
};
export default ShowProfileActions;
