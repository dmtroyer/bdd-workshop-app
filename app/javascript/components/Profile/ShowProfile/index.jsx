import React from 'react';
import { Row, Col } from 'reactstrap';
import PropTypes from 'prop-types';
import Avatar from '../../Avatar';

const ShowProfile = ({ profile }) => {
  return (
    <>
      <Avatar profile={profile} />
      <Row>
        <Col>
          <dl>
            <dt>handle</dt>
            <dd data-testid="details-handle">{profile.handle}</dd>
            <dt>email</dt>
            <dd data-testid="details-email">{profile.email}</dd>
          </dl>
        </Col>
      </Row>
    </>
  );
};
ShowProfile.propTypes = {
  profile: PropTypes.shape({
    id: PropTypes.string.isRequired,
    handle: PropTypes.string,
    email: PropTypes.string,
  }),
};

export default ShowProfile;
