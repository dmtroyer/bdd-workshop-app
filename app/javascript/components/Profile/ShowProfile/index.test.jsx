import React from 'react';
import { shallow } from 'enzyme';
import ShowProfile from '.';

describe('ShowProfile', () => {
  it('Displays avatar, handle and email', () => {
    const wrapper = shallow(
      <ShowProfile
        profile={{
          id: 'the-id',
          handle: 'the-handle',
          email: 'princess@email.com',
        }}
      />
    );
    expect(wrapper.find('dl')).toMatchInlineSnapshot(`
<dl>
  <dt>
    handle
  </dt>
  <dd
    data-testid="details-handle"
  >
    the-handle
  </dd>
  <dt>
    email
  </dt>
  <dd
    data-testid="details-email"
  >
    princess@email.com
  </dd>
</dl>
`);
  });
});
