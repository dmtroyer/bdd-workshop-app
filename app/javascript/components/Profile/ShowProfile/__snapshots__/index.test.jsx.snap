// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowProfile Displays avatar, handle and email 1`] = `
<Fragment>
  <Avatar
    profile={
      Object {
        "email": "princess@email.comn",
        "handle": "the-handle",
        "id": "the-id",
      }
    }
  />
  <Row
    tag="div"
  >
    <Col
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <dl>
        <dt>
          handle
        </dt>
        <dd
          data-testid="details-handle"
        >
          the-handle
        </dd>
      </dl>
      <dl>
        <dt>
          email
        </dt>
        <dd
          data-testid="details-email"
        >
          princess@email.comn
        </dd>
      </dl>
    </Col>
  </Row>
</Fragment>
`;
